draft-js-plugins-editor/src/Editor/index.js

```diff
@@ -63,14 +63,17 @@ class PluginEditor extends Component {
   }
 
   componentWillReceiveProps(next) {
-    if (
-      this.props.editorState.getDecorator() !== null &&
-      this.props.editorState.getDecorator() !== next.editorState.getDecorator()
-    ) {
-      const decorator = this.props.editorState.getDecorator();
-      const editorState = EditorState.set(next.editorState, { decorator });
-      this.onChange(moveSelectionToEnd(editorState));
-    }
+    const curr = this.props;
+    const currDec = curr.editorState.getDecorator();
+    const nextDec = next.editorState.getDecorator();
+
+    if (currDec === nextDec) return;
+    if (currDec && nextDec && currDec.decorators.size === nextDec.decorators.size) return;
+    if (!currDec && nextDec) return;
```

Comment:
so this says - if the current decorator is null and the next exists we return and don't change decorators? Uhm, why? Can we remove this?

Reply:
Because then it's already in the next editor state, lol. So if you then call `onChange` you get a loop.

Reply:
Oh wait, this could actually be simplified. One sec